flybywiresim/a32nx
# feat(fpm, lnav): refactoring of flight path building / rendering. add support for VM legs

## Summary of Changes

This is an initial refactoring of leg building and flight path rendering.

* Legs are now declared in `guidance/lnav/legs`.
* Multiple leg geometry now generates transitions (still hard-coded to Type I) for all legs in the flight plan.
* This allows us to use a simple reverse-iteration instead of having different / repeated logic for different legs or transitions.
* Leg and transition indices in Geometry are now different. Transition at index n is now the transition after leg at index n.
* The flight plan rendering code has been cleaned up to match the reverse iteration for geometry generation, and not contain special edge cases for first / second leg / transition.

Additionally, the following changes have been introduced to help the refactor:

* VM legs are now generated from waypoints for which `isVectors = true`.
  * [x] Rendering
  * [x] Giodamce

* [x] Use correct heading for VM legs
* [x] Fix rendering of active VM legs 

In the future, the plans are to introduce lower-level path primitives to use for both guidance and drawing. This will introduce proper vectorization.

Discord username (if different from GitHub): someperson#4953

## Testing Instructions

* Load in a SID containing a MANUAL leg (for example, CYYZ06R AVSEP6)
* Make sure NAV mode keeps infinite line heading without sequencing leg until DIR TO
* Perform DIR TO to downstream waypoint
* Observe correct sequencing


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
